var/spack/repos/builtin/packages/cosma/package.py

```diff
@@ -7,61 +7,92 @@
 from spack import *
 
 
-class Cosma(CMakePackage, CudaPackage):
+class Cosma(CMakePackage):
     """
     Distributed Communication-Optimal Matrix-Matrix Multiplication Library
     """
 
     maintainers = ['haampie', 'kabicm', 'teonnik']
     homepage = 'https://github.com/eth-cscs/COSMA'
-    url = 'https://github.com/eth-cscs/COSMA/releases/download/v2.2.0/cosma.tar.gz'
+    url = 'https://github.com/eth-cscs/COSMA/releases/download/v2.5.1/COSMA-v2.5.1.tar.gz'
     git = 'https://github.com/eth-cscs/COSMA.git'
 
+    def url_for_version(self, version):
+        if version <= Version('2.3.0'):
+            return 'https://github.com/eth-cscs/COSMA/releases/download/v{0}/cosma.tar.gz'.format(version)
+
+        return 'https://github.com/eth-cscs/COSMA/releases/download/v{0}/COSMA-v{1}.tar.gz'.format(version, version)
+
     # note: The default archives produced with github do not have the archives
     #       of the submodules.
     version('master', branch='master', submodules=True)
+    version('2.5.1', sha256='085b7787597374244bbb1eb89bc69bf58c35f6c85be805e881e1c0b25166c3ce')
+    version('2.5.0', sha256='7f68bb0ee5c80f9b8df858afcbd017ad4ed87ac09439d13d7d890844dbdd3d54')
+    version('2.4.0', sha256='5714315ce06d48037f86cfee2d7f19340643fee95e9d7f1e92dc1b623b67e395')
     version('2.3.0', sha256='0c01c2deb5a0cd177952178350188a62c42ce55e604d7948ac472f55bf0d4815')
     version('2.2.0', sha256='1eb92a98110df595070a12193b9221eecf9d103ced8836c960f6c79a2bd553ca')
     version('2.0.7', sha256='8d70bfcbda6239b6a8fbeaca138790bbe58c0c3aa576879480d2632d4936cf7e')
     version('2.0.2', sha256='4f3354828bc718f3eef2f0098c3bdca3499297497a220da32db1acd57920c68d')
 
-    variant('scalapack', default=False,
-            description='Build with ScaLAPACK API.')
+    # We just need the libraries of cuda and rocm, so no need to extend
+    # CudaPackage or ROCmPackage.
+    variant('cuda', default=False, description='Build with cuBLAS support')
+    variant('rocm', default=False, description='Build with rocBLAS support')
```

Comment:
Can you please elaborate on why there is no need to extend the built-in base packages?

Reply:
Hey Tamara, the reason is that cosma just needs the libraries to link to, but not the CUDA or HIP language or compilers for those.

Reply:
Okay. But inheriting from `CudaPackage` and `ROCmPackage` gives you access to the standard variants, known/supported targets, version- and arch-specific dependencies, and known conflicts.

Reply:
BUT I have been schooled on why it doesn't matter for you. :wink: